kimchi_state/src/kimchi_state_server.cpp

```diff
@@ -208,7 +208,6 @@ void KimchiStateServer::initialPoseCallback(
   locate_thread.detach();
   response->success = true;
 
-  changeState(RobotState::IDLE);
```

Comment:
Reasoning: the change to IDLE has to be done when the robot is already localized. Given that this method is not waiting for the thread to finish, setting the state here is doing it before the robot is localized

Reply:
Nice catch! You are right, I already introduced a fix for this on the following PR as well.